bfvmm/src/hve/arch/intel_x64/phys_mtrr.cpp

```diff
@@ -0,0 +1,279 @@
+//
+// Bareflank Hypervisor
+// Copyright (C) 2018 Assured Information Security, Inc.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+#include <array>
+#include <cstdio>
+#include <arch/x64/misc.h>
+#include <arch/intel_x64/bit.h>
+#include <arch/intel_x64/mtrr.h>
+#include <hve/arch/intel_x64/mtrr.h>
+#include <hve/arch/intel_x64/phys_mtrr.h>
+
+namespace eapis
+{
+namespace intel_x64
+{
+
+using namespace ::intel_x64::mtrr;
+
+///
+/// Phys MTRR
+///
+phys_mtrr::phys_mtrr()
+{
+    expects(::intel_x64::mtrr::is_supported());
+
+    const auto cap = ia32_mtrrcap::get();
+    const auto num = ia32_mtrrcap::vcnt::get(cap);
+    expects(num > 0U && num <= 0xFFU);
+
+    const auto def = ia32_mtrr_def_type::get();
+    expects(ia32_mtrr_def_type::e::is_enabled(def));
+
+    m_cap = cap;
+    m_def = def;
+    m_pas = ::x64::cpuid::addr_size::phys::get();
+
+    this->init_fixed_ranges();
+    this->init_variable_ranges();
+}
+
+static uint64_t fixed_range_size(uint64_t range)
+{
+    expects(range <= 87U);
+
+    if (range <= 7U) {
+        return 1U << 16U;
+    }
+    else if (range <= 23U) {
+        return 1U << 14U;
+    }
+
+    return 1U << 12U;
+}
+
+void
+phys_mtrr::print_variable_ranges(uint64_t level) const
```

Comment:
This private method doesn't get called by anything. Wondering if you left this in here by accident?